src/main/java/org/candlepin/model/DeletedConsumerCurator.java

```diff
@@ -103,19 +123,93 @@ public int createDeletedConsumers(Collection<String> consumerIds) {
             .executeUpdate();
     }
 
-    @SuppressWarnings("unchecked")
-    public List<DeletedConsumer> findByDate(Date date) {
-        EntityManager em = this.getEntityManager();
-        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+    /**
+     * Fetches a collection of deleted consumers based on the data in the query builder. If the
+     * query builder is null or contains no arguments, the query will not limit or sort the result.
+     *
+     * @param queryArgs
+     *     a DeletedConsumerQueryArguments instance containing the various arguments to use to
+     *     select contents
+     *
+     * @return a list of deleted consumers. It will be paged and sorted if specified
+     */
+    public List<DeletedConsumer> listAll(DeletedConsumerQueryArguments queryArgs) {
+        CriteriaBuilder criteriaBuilder = this.getEntityManager().getCriteriaBuilder();
         CriteriaQuery<DeletedConsumer> criteriaQuery = criteriaBuilder.createQuery(DeletedConsumer.class);
-        Root<DeletedConsumer> consumer = criteriaQuery.from(DeletedConsumer.class);
-        criteriaQuery.select(consumer);
 
-        criteriaQuery.where(criteriaBuilder.greaterThanOrEqualTo(
-            consumer.get(DeletedConsumer_.CREATED), date));
-        criteriaQuery.orderBy(criteriaBuilder.desc(consumer.get(DeletedConsumer_.CREATED)));
+        Root<DeletedConsumer> root = criteriaQuery.from(DeletedConsumer.class);
+        criteriaQuery.select(root)
+            .distinct(true);
+        if (queryArgs.getDate() != null) {
+            criteriaQuery.where(criteriaBuilder.greaterThanOrEqualTo(
+                root.get(DeletedConsumer_.CREATED),
+                new Date(queryArgs.getDate().toInstant().toEpochMilli())));
+        }
 
-        return em.createQuery(criteriaQuery)
-            .getResultList();
+        List<Predicate> predicates = new ArrayList<>();
+        Predicate securityPredicate = this.getSecurityPredicate(DeletedConsumer.class, criteriaBuilder,
+            root);
+        if (securityPredicate != null) {
+            predicates.add(securityPredicate);
+        }
+
+        if (!predicates.isEmpty()) {
+            criteriaQuery.where(predicates.toArray(new Predicate[predicates.size()]));
+        }
```

Comment:
Since the predicates list will be one element at max we can just check if securityPredicate is not null and call `criteriaQuery` if it is not null. This avoids creating a list and then moving the element from the list to an array.

```
Predicate securityPredicate = this.getSecurityPredicate(DeletedConsumer.class, criteriaBuilder,
    root);
if (securityPredicate != null) {
    criteriaQuery.where(securityPredicate);
}
```

This can also be applied in the `getDeletedConsumerCount` method.